src/components/organisms/details-collapsible/index.tsx

```diff
@@ -0,0 +1,26 @@
+import * as RadixCollapsible from "@radix-ui/react-collapsible"
+import clsx from "clsx"
+import React, { useState } from "react"
+import ArrowDownIcon from "../../fundamentals/icons/arrow-down-icon"
+import ArrowUpIcon from "../../fundamentals/icons/arrow-up-icon"
+
+const DetailsCollapsible = ({ children }) => {
+  const [open, setOpen] = useState(false)
+
+  const Icon = open ? ArrowUpIcon : ArrowDownIcon
+  const label = open ? "Hide additional details" : "Show additional details"
+
+  return (
+    <RadixCollapsible.Root onOpenChange={(state) => setOpen(state)}>
+      <RadixCollapsible.Trigger className={clsx("ml-4", { ["mb-6"]: open })}>
```

Comment:
We should probably remove the `ml-4` from the trigger and potentially add it through a `triggerClassname` property instead to give more flexibility in terms of styling

(i.e. in the manage gift card pr the margin should be ml-2 to match the design)